### PR TITLE
Pr rocm6

### PR DESCRIPTION
**Pull request by gmerlo**
Minor fixes necessary for rocm>6

### CODE REVIEW

**Review by germasch — APPROVED** [review]
LGTM.

I'm not much of a fan of the `static_cast<void>`, but I suppose you got warnings (or worse) because of the ignored return value -- did you consider actually checking the return value? I usually just do something like

```
rc = hipFree(...);
assert(rc == 0) // or whatever the success return is....
```
Of course, this doesn't actually handle failure either...

File: src/fortran/gpu_api.cxx [hunk 1]
```diff
@@ -88,7 +88,7 @@ extern "C" void gpuDeviceReset()
 #ifdef GTENSOR_DEVICE_CUDA
   cudaDeviceReset();
 #elif defined(GTENSOR_DEVICE_HIP)
-  hipDeviceReset();
```
Comment (bd4) [inline src/fortran/gpu_api.cxx:91]: This should probably be gtCheck, we don't really want to ingore the return value.

File: include/gtensor/backend_hip.h [hunk 2]
```diff
@@ -268,19 +272,27 @@ class backend_ops<gt::space::hip>
     hipPointerAttribute_t attr;
     auto rc = hipPointerGetAttributes(&attr, ptr);
     if (rc == hipErrorInvalidValue) {
-      hipGetLastError(); // clear the error
+      static_cast<void>(hipGetLastError()); // clear the error
```
Comment (bd4) [inline include/gtensor/backend_hip.h:275]: Here we are purposely ignoring the return val to clear the error. I think C style (void) is easier to read, but don't have a strong opinion.

File: include/gt-solver/backend/hip-csrsm2.h [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 #include "gt-blas/blas.h"
 
-#include <rocsparse.h>
+#include <rocsparse/rocsparse.h>
```
Comment (bd4) [inline include/gt-solver/backend/hip-csrsm2.h:12]: does this still work with Rocm 5.x? I think we can drop support for 4.x now, but not sure we want to drop 5 yet.

**Review by bd4 — CHANGES_REQUESTED** [review]

**Comment by bd4** [comment]
CI failure is unrelated, I guess we were pulling dev branch of umpire and it wants newer cmake.

**Comment by gmerlo** [comment]
this was me mostly being lazy and avoiding warnings. There is however a major regression with rocm6, as if the memory is not correctly allocated anymore. But could also be something else, not sure yet what is going so I'll keep this a draft for a little longer

**Comment by bd4** [comment]
merged same feature with backwards compat for <5.5 in #292

**Pull request closed without merging**